lib/blocks.php

```diff
@@ -45,3 +45,71 @@ function gutenberg_reregister_core_block_types() {
 	}
 }
 add_action( 'init', 'gutenberg_reregister_core_block_types' );
+
+/**
+ * Registers a new block style.
+ *
+ * @param string $block_name       Block type name including namespace.
+ * @param array  $style_properties Array containing the properties of the style name, label, style (name of the stylesheet to be enqueued), inline_style (string containing the CSS to be added).
+ *
+ * @return boolean True if the block style was registered with success and false otherwise.
+ */
+function register_block_style( $block_name, $style_properties ) {
+	return WP_Block_Styles_Registry::get_instance()->register( $block_name, $style_properties );
+}
+
+/**
+ * Unregisters a block style.
+ *
+ * @param string $block_name       Block type name including namespace.
+ * @param array  $block_style_name Block style name.
+ *
+ * @return boolean True if the block style was unregistered with success and false otherwise.
+ */
+function unregister_block_style( $block_name, $block_style_name ) {
+	return WP_Block_Styles_Registry::get_instance()->unregister( $block_name, $block_style_name );
+}
+
+/**
+ * Function responsible for enqueuing the assets required for block styles functionality.
+ */
+function enqueue_block_styles_assets() {
+	$block_styles = WP_Block_Styles_Registry::get_instance()->get_all_registered();
+
+	foreach ( $block_styles as $block_name => $styles ) {
+		foreach ( $styles as $style_properties ) {
+			if ( isset( $style_properties['style_handle'] ) ) {
+				wp_enqueue_style( $style_properties['style_handle'] );
+			}
+			if ( isset( $style_properties['inline_style'] ) ) {
+				wp_add_inline_style( 'wp-block-library', $style_properties['inline_style'] );
+			}
+
+			$client_style_properties = $style_properties;
+			if ( isset( $client_style_properties['style_handle'] ) ) {
+				unset( $client_style_properties['style_handle'] );
+			}
+			if ( isset( $client_style_properties['inline_style'] ) ) {
+				unset( $client_style_properties['inline_style'] );
+			}
+
+			wp_add_inline_script(
```

Comment:
I feel this should be in a separate action `enqueue_block_editor_assets` that is specific to the editor (the styles are good right now though). We could also avoid `wp_add_inline_script` and enqueue a separate `inline script` that depends on `wp-blocks` instead.

Reply:
Hi @youknowriad, I updated to use another action enqueue_block_editor_assets for the addition of the script as suggested 👍 

Regarding the way to avoid wp_add_inline_script, I'm probably missing something -- I'm not seeing how we can avoid a call to wp_add_inline_script (unless printing our own inline script tags).
We can do something similar to:
```
	wp_register_script( 'wp-block-styles', false, array( 'wp-blocks' ) );
	wp_add_inline_script( 'wp-block-styles', $inline_script );
	wp_enqueue_script( 'wp-block-styles' );
```
Registering our own script that does not point to any file, but that seems more hacky and I'm not sure if phpcs will not complain of this approach.

In this approach, we still have a call to wp_add_inline_script, was your idea different from what we have or from this approach I commented?

I performed another side update; now, all the scripts are registered together in a single function and a single inline script. This change saves some bytes on every editor page load.
For the inline styles, it is better not to apply a similar change. Inline-styles are not generated by us they are provided by the plugin/theme developers, having an invalid style in a plugin would invalid other styles.

Reply:
> Registering our own script that does not point to any file, but that seems more hacky and I'm not sure if phpcs will not complain of this approach.

This is exactly what I was suggesting. The advantage is that no matter if `wp-blocks` is already enqueued or not, it will work (no race conditions).

Reply:
I don't see it as crucial though.